Core/include/Acts/Seeding/Seedfinder.hpp

```diff
@@ -57,17 +61,25 @@ class Seedfinder {
   /// Ranges must return pointers.
   /// Ranges must be separate objects for each parallel call.
   /// @return vector in which all found seeds for this group are stored.
-  template <typename sp_range_t>
-  std::vector<Seed<external_spacepoint_t>> createSeedsForGroup(
-      sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs) const;
+//template< typename sp_range_t >
+  template< typename T=Platform_t, typename sp_range_t>
+  typename std::enable_if< std::is_same<T, Acts::CPU>::value, std::vector<Seed<external_spacepoint_t> > >::type
+  createSeedsForGroup(sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs) const;
+
+#ifdef  ACTS_HAS_CUDA
+    
+  template< typename T=Platform_t, typename sp_range_t>
+  typename std::enable_if< std::is_same<T, Acts::CUDA>::value, std::vector<Seed<external_spacepoint_t> > >::type
+  createSeedsForGroup(sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs) const;
+
+  std::tuple< double, double, double, double > getTimeMetric() { return t_metric; }
 
+#endif
+    
  private:
-  void transformCoordinates(
-      std::vector<const InternalSpacePoint<external_spacepoint_t>*>& vec,
-      const InternalSpacePoint<external_spacepoint_t>& spM, bool bottom,
-      std::vector<LinCircle>& linCircleVec) const;
 
   Acts::SeedfinderConfig<external_spacepoint_t> m_config;
+  mutable std::tuple< double, double, double, double > t_metric; // doublet search, transform coordinate, triplet search, wall time
```

Comment:
Not entirely sure how this is used, but `mutable` is typically a very bad sign for thread-safety.

Reply:
It is just for measuring elapsed time for each part of seed finding algorithm.
If people think it is an appropriate feature, it can be removed before the PR is accpeted.

Reply:
If this is ever used concurrently from more than one thread, this will/would cause crashes. If you can guarantee this won't happen, it would be acceptable, but we usually don't use `mutable` anywhere in the code. In principle, the feature is fine, you'd just have to put this mutable variable in the state I guess.

Reply:
I wouldn't add code to measure performance into the repo, especially code measuring substeps of the algorithm. Additionally, the measurements would be rather inaccurate as multiple threads could simultaneously (over)write their respective results

Reply:
Okay I will remove them before the PR is merged

Reply:
I just removed them!